### PR TITLE
rz-svd: Add `yxml` as a subproject

### PR DESCRIPTION
**Pull request by officialcjunior**
* Adds `yxml` as a subproject for rz-svd
* Get a CI and add a job for `rz-svd`

Partially addresses: https://github.com/rizinorg/rizin-extras/issues/9

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,56 +1,34 @@
...
-        export R2PM_GITSKIP=1
-        for p in $R2PIPE_TESTS ; do
-          echo $p
-          r2pm -i $p
-        done
-        set +e
-
+          cd rz-svd
+          export LD_LIBRARY_PATH=${HOME}/lib/$(uname -m)-linux-gnu:${HOME}/lib:${HOME}/lib64:${LD_LIBRARY_PATH}
+          export PKG_CONFIG_PATH=${HOME}/lib/pkgconfig:${HOME}/lib/$(uname -m)-linux-gnu/pkgconfig:${PKG_CONFIG_PATH}
```
Comment (ret2libc) [inline .github/workflows/ci.yml]: Are these necessary given that you have installed rizin on /usr? I think they shouldn't be.

File: rz-svd/meson.build [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+  meson_version: '>=0.55.0',
+  default_options: [
+    'buildtype=debugoptimized',
+    'b_vscrt=from_buildtype',
+  ]
+)
 
+# handle the yxml dependency
 plugins_dir = join_paths(get_option('prefix'), 'share', 'rizin', 'plugins')
+check_meson_subproject_py = files('sys/check_meson_subproject.py')
```
Comment (ret2libc) [inline rz-svd/meson.build]: ignore this... I think we don't need such level of user-friendliness yet for these. Just remove the whole check_meson_subproject stuff for now.

File: rz-svd/meson.build [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
+# handle the yxml dependency
 plugins_dir = join_paths(get_option('prefix'), 'share', 'rizin', 'plugins')
 
+py3_exe = import('python').find_installation()
+
+r = run_command(py3_exe, 'yxml')
+if r.returncode() == 1 and get_option('subprojects_check')
+  error('Subprojects are not updated. Please run `git clean -dxff subprojects/` to delete all local subprojects directories. If you want to compile against current subprojects then set option `subprojects_check=false`.')
+endif
+
```
Comment (ret2libc) [inline rz-svd/meson.build]: remove

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -1,56 +1,32 @@
...
-    - name: Init r2pm
-      run: r2pm init
-    - name: Compile and install plugins
+          git clone --depth=1 https://github.com/rizinorg/rizin
+          cd rizin
+          meson --buildtype=release --prefix=/usr build
+          ninja -C build
+          sudo ninja -C build install
+          cd ..
+    - name: Install rz-svd
```
Comment (ret2libc) [inline .github/workflows/ci.yml:26]: You could add another step after this which runs rz-asm -L | grep rz-svd (or whatever) to really check if rizin loads the plugin correctly.

Comment (officialcjunior) [reply .github/workflows/ci.yml:26]: Good idea.
This thing is a core plugin and gets displayed when you run `Lc`, so it doesn't show up on `rz-asm` I believe.
Let me see if there's another way to get it on the screen.

Comment (ret2libc) [reply .github/workflows/ci.yml:26]: if so, do `rizin -qc "Lc~rz-svd"`

Comment (officialcjunior) [reply .github/workflows/ci.yml:26]: Okay, that seems to work.
But how do we do it in the CI? Would just running the above command be enough?

Comment (officialcjunior) [reply .github/workflows/ci.yml:26]: Okay, seems to be working fine on the CI now:

```bash
Installing librz-svd.so to /home/runner/.local/share/rizin/plugins
rz-svd: SVD Plugin for Rizin (Made by officialcjunior, v0.1.0, LGPL3)
```

**Review by ret2libc — APPROVED** [review]

**Comment by ret2libc** [comment]
Can we merge?

**Review by XVilka — APPROVED** [review]
Good. Though I think we should move `yxml` into `RzUtil` in Rizin and provide some wrappers for the public API, like we have for the JSON. Might be useful in many multiple Rizin subsystems and plugins.

**Pull request merged**